docs/devel/releasing.md

```diff
@@ -1,7 +1,155 @@
+of cherry-picks. To create a candidate cherry-pick release and validate it, do
+the following:
+
+1. `export VER=x.y` (e.g. `0.20` for v0.20 - we don't care about the patch level
+   yet, we're just working on the release branch)
+1. `git checkout -b release-${VER} upstream/release-${VER}` (**NOTE:** This
+   assumes all release steps were followed for the `.0` release, which should
+   result in a `release-x.y` branch. If not, this step is recoverable, but stop
+   and ask someone.)
+1. Make sure you don't have any files you care about littering your repo (they
+   better be checked in or outside the repo, or the next step will delete them).
+1. `make clean && git reset --hard HEAD && git clean -xdf`
+1. `git cherry-pick -m 1 <merge commit of pull>` for each thing you need cherry
+   picked into the patch release. **NOTE**: Read carefully. You're selecting the
+   hash of the merge commit on `master`
+   (e.g. [974377b3](https://github.com/GoogleCloudPlatform/kubernetes/commit/974377b3064ac59b6e5694bfa568d67128026171)). This
+   keeps the PR #'s on the release branch for easy verification (at the mild
+   expense of squashing original authorship).
+1. Build and run e2e's: `go run hack/e2e.go -v -build -up -test -down |& tee /tmp/log`
+
+Assuming all that went well, you now have a candidate set of patches in your
+branch. **NOTE**: The binaries you just produced aren't official. Don't try to
+push them. (We haven't marked them with the new version or anything.)
+
+#### Tagging and Merging
+
+Do the following (you still have `${VER}` set and you're still working on the
+`release-${VER}` branch, right?):
+
+1. `export PATCH=z` where `z` is the patch level of `vx.y.z`
+1. `./build/mark-new-version.sh v${VER}.${PATCH}` to mark the new release and get further
+   instructions. This creates a series of commits on the branch you're working
+   on (`release-${VER}`), including forking our documentation for the release,
+   the release version commit (which is then tagged), and the post-release
+   version commit.
+1. Follow the instructions given to you by that script. They are canon for the
+   remainder of the Git process. If you don't understand something in that
+   process, please ask!
+
+**TODO**: how to fix tags, etc., if the release is changed.
+
+#### Building and Pushing Binaries
+
+In your git repo (you still have `${VER}` and `${PATCH}` set from above right?):
+
+1. `git checkout upstream/master && build/build-official-release.sh
+   v${VER}.${PATCH}` (the `build-official-release.sh` script is version
+   agnostic, so it's best to run it off `master` directly).
+1. Follow the instructions given to you by that script. At this point, you've
+   done all the Git bits, you've got all the binary bits pushed, and you've got
+   the template for the release started on GitHub.
+
+#### Writing Release Notes
+
+Release notes for a patch release are fast. You just did all the cherry-picks by
+squashing pull requests, so you have the PR #'s in `git log
```

Comment:
I was trying to describe what `git cherry-pick -m` does, which is kind of a weird beast. It picks up the merge pull and patches in the resulting diff, which I think is a better/less-error-prone cherry-pick procedure than individual commit cherry-picks (because a pull request on master might've brought in several commits), but it's a squashed commit of sorts. I'll try to reword that.